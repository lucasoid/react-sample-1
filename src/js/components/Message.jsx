import React from 'react';

export default class Message extends React.Component {
    
    render() {
        var date = new Date(this.props.timestamp);
        var addClass = this.props.me === true ? 'me' : 'not-me';
        return (
            <div className={'message ' + addClass}>
                <div className="user">{this.props.user}:</div>
                <div className="msg"><span>{this.props.msg == '' ? '...' : this.props.msg}</span></div>
                <div className="date">{date.toLocaleString()}</div>
            </div>
        );
    }
}