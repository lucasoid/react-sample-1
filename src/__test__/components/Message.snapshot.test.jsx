/* Snapshot render example */

import React from 'react';
import renderer from 'react-test-renderer';
import Message from '../../js/components/Message';

test('Message is populated with user, msg, and date values', function() {
    
    const timestamp = 1487278301187;
    
    const component = renderer.create(
        <Message me={false} user="Boots" msg="Meow" timestamp={timestamp} />
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});