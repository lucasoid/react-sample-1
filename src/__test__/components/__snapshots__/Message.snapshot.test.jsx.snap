exports[`test Message is populated with user, msg, and date values 1`] = `
<div
  className="message not-me">
  <div
    className="user">
    Boots
    :
  </div>
  <div
    className="msg">
    <span>
      Meow
    </span>
  </div>
  <div
    className="date">
    2/16/2017, 3:51:41 PM
  </div>
</div>
`;
